queue_services/entity-auth/src/entity_auth/resources/worker.py

```diff
@@ -232,22 +243,45 @@ def create_affiliation(legal_entity: LegalEntity, filing: Filing):
         raise AccountServiceException from err
 
 
-def get_firm_affiliation_passcode(legal_entity_id: int):
+def get_firm_affiliation_passcode(business: AlternateName | LegalEntity):
     """Return a firm passcode for a given business identifier."""
     pass_code = None
-    end_date = datetime.utcnow().date()
-    entity_roles = EntityRole.get_entity_roles(legal_entity_id, end_date)
+    end_date = datetime.now(timezone.utc).date()
 
-    if len(entity_roles) == 0:
-        return pass_code
+    if business.is_alternate_name_entity and business.entity_type == BusinessCommon.EntityTypes.SOLE_PROP.value:
+        if business.is_owned_by_legal_entity_person:
+            pass_code = business.legal_entity.last_name + ", " + business.legal_entity.first_name
+            if business.legal_entity.middle_initial:
+                pass_code = pass_code + " " + business.legal_entity.middle_initial
+        elif business.is_owned_by_legal_entity_org or business.is_owned_by_colin_entity:
+            pass_code = business.legal_name
 
-    party = entity_roles[0].related_entity
+    elif entity_roles := EntityRole.get_entity_roles(business.id, end_date):
+        party = entity_roles[0].related_entity
```

Comment:
for the scenario where the partner is in COLIN, entity_roles[0].related_colin_entity will be populated.  And the entry will linked to `colin_entities` table.